ecommerce/api.py

```diff
@@ -242,18 +243,26 @@ def refund_order(*, order_id: int, **kwargs):
     unenroll = kwargs.get("unenroll", False)
 
     with transaction.atomic():
-
-        order = FulfilledOrder.objects.select_for_update().get(id=order_id)
-
+        if reference_number is not None:
+            order = FulfilledOrder.objects.select_for_update().get(
+                reference_number=reference_number
+            )
+        elif order_id is not None:
+            order = FulfilledOrder.objects.select_for_update().get(pk=order_id)
+        else:
+            log.debug(
+                f"Either order_id or reference_number is required to fetch the Order. "
+            )
+            return False
         if order.state != Order.STATE.FULFILLED:
-            log.debug(f"Order with order_id {order_id} is not in fulfilled state.")
+            log.debug(f"Order with order_id {order.id} is not in fulfilled state.")
```

Comment:
For this and the other log strings, it should use `%s` formatting so they get deduplicated in sentry.